dotCMS/src/main/java/com/dotmarketing/util/ImportStarterUtil.java

```diff
@@ -0,0 +1,1148 @@
+import com.dotmarketing.beans.Tree;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.business.DuplicateUserException;
+import com.dotmarketing.business.Layout;
+import com.dotmarketing.business.LayoutsRoles;
+import com.dotmarketing.business.PortletsLayouts;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.business.UsersRoles;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.db.HibernateUtil;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.logConsole.model.LogMapperRow;
+import com.dotmarketing.portlets.containers.model.Container;
+import com.dotmarketing.portlets.contentlet.business.Contentlet;
+import com.dotmarketing.portlets.folders.business.FolderAPI;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.portlets.languagesmanager.model.Language;
+import com.dotmarketing.portlets.links.model.Link;
+import com.dotmarketing.portlets.rules.util.RulesImportExportUtil;
+import com.dotmarketing.portlets.structure.model.Relationship;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.portlets.workflows.util.WorkflowImportExportUtil;
+import com.dotmarketing.startup.runalways.Task00004LoadStarter;
+import com.liferay.portal.SystemException;
+import com.liferay.portal.ejb.CompanyManagerUtil;
+import com.liferay.portal.model.Company;
+import com.liferay.portal.model.Image;
+import com.liferay.portal.model.User;
+import com.liferay.util.Base64;
+import com.liferay.util.Encryptor;
+import com.liferay.util.EncryptorException;
+import com.liferay.util.FileUtil;
+import io.vavr.control.Try;
+
+
+/**
+ * This utility is part of the {@link Task00004LoadStarter} task, which fills the empty dotCMS
+ * tables on a fresh install with information regarding the Demo Site that the application ships
+ * with. This allows users to be able to log into the dotCMS back-end and interact with the system
+ * before adding their own custom content.
+ *
+ * @author Jason Tesser
+ * @version 1.6
+ *
+ */
+public class ImportStarterUtil {
+
+    /**
+     * The path where tmp files are stored. This gets wiped alot
+     */
+    private String backupTempFilePath = ConfigUtils.getBackupPath() + File.separator + "temp";
+    private ArrayList<String> classesWithIdentity = new ArrayList<String>();
+    private Map<String, String> sequences;
+    private Map<String, String> tableIDColumns;
+    private Map<String, String> tableNames;
+    private final String assetPath;
+
+
+    private final List<File> tempFiles;
+
+    private static final String SYSTEM_FOLDER_PATH = FolderAPI.SYSTEM_FOLDER_PARENT_PATH;
+
+    private final File starterZip;
+
+
+    public ImportStarterUtil(File starterZipFile) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'starterZipFile' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)